File: dart/areTheyTheSame.dart
import "dart:math";

bool comp(List<int> a1, List<int> a2) {
  int c = 0;
  if (a1.length == a2.length) {
    for (int i in a1) {
      if (a2.contains(pow(i, 2))) {
        a2.remove(pow(i, 2));
        c++;
      } else {
        return false;
      }
    }
  }
  if (c == a1.length) {
    return true;
  }
  return false;
}

File: python/areTheyTheSame.py
def comp(a1, a2):
    if a1 is None or a2 is None:
        return False

    elif len(a1) == len(a2):
        for i in a1:
            if i*i in a2:
                a2.remove(i*i)
            
    if len(a2) == 0:
        return True
    return False
